src/transformers/modeling_bart.py

```diff
@@ -360,16 +382,20 @@ def forward(
         )
         x = F.dropout(x, p=self.dropout, training=self.training)
         x = residual + x
+        if not self.normalize_before:
```

Comment:
Not a big fan of the `self.normalize_before `and after statements. Seems hard to avoid though. Maybe Bart could use some `BartAttention` and `BartIntermediate` Layer classes similar to Bert's implementation and then a much smaller `BartEncoder` and `BartMultiLingualEncoder` Layer which uses the `BartAttention` and `BartIntermediate` layers and puts the LayerNorm differently in the correct place. But I guess that would also require to change the way the weights are loaded into the model, so not ideal either ...

Reply:
Yeah I spent an hour trying to move it around. 

Also tried to dynamically define `nn.ModuleList` with layer_norm in the correct place.

This felt more readable.